example/test-hmap.cc

```diff
@@ -34,6 +35,49 @@ int main(int argc, const char* argv[]) {
 		/////////////////////////////////////////////////
 		
 	}
+	{
+		class Foo {
+		private:
+			const std::string& bar_;
+		public:
+			explicit Foo(const std::string& bar) : bar_(bar) {}
+			const std::string& str() const { return bar_; }
+		};
+		auto myMap = make_dynamic_hmap((dK<Foo>("baz"), Foo("Listen"s)),
+		                               (dK<Foo>("bobndoug"), Foo("hosers"s)),
+		                               (dK<Foo>("foo"), Foo("I say"s)));
+
+		// Since this is a citation of Herman's Hermits, not of
+		// Bob and Doug of SCTV, replace "hosers" with "people"
+		auto [iter, inserted] =
+		    myMap.insert_or_assign(dK<Foo>("bobndoug"), "people"s);
+		assert(!inserted); // Assert that "people" was assigned, not inserted
+		auto [iter2, inserted2] =
+		    myMap.insert_or_assign(dK<Foo>("cusp"), "to what"s);
+		assert(inserted2); // Assert that "to what" was inserted
+
+		std::cout << myMap.at(dK<Foo>("baz")).str() << ' ';
+		std::cout << iter->second.str() << ' ';
+		std::cout << iter2->second.str() << ' ';
+		std::cout << myMap.at(dK<Foo>("foo")).str() << std::endl;
+
+		myMap = make_dynamic_hmap((dK<Foo>("bobndoug"), Foo("Everybody's"s)),
+		                          (dK<Foo>("cusp"), Foo("somebody"s)),
+		                          (dK<Foo>("foo"), Foo("sometime"s)));
+
+		auto [iter3, inserted3] =
+		    myMap.try_emplace(dK<Foo>("bobndoug"), "All the hosers"s);
+		assert(!inserted3); // Assert that "All the hosers" went nowhere, eh
+
+		auto [iter4, inserted4] =
+		    myMap.try_emplace(dK<Foo>("crack"), "got to love"s);
```

Comment:
I would perhaps prefer that we keep our test cases free of drug references. Perhaps `"maplesyrup"` or `"timhortons"` is an appropriately Canadian substitution for your theme above.

Reply:
I changed it from "crack" to "chair".  I was trying to pick a key word that began with "c" and had no drug allusions in mind.  Bob and Doug are hosers, but they stick to Canadian beer (which is Canadian malt liquor for us Americans).  They sell the American version in Canada too (it's called "export"... e.g., "Molson Export").